seqr/views/apis/igv_api_tests.py

```diff
@@ -18,26 +18,35 @@
 class IgvAPITest(AuthenticationTestCase):
     fixtures = ['users', '1kg_project']
 
-    @mock.patch('seqr.utils.file_utils.subprocess.Popen')
-    def test_proxy_google_to_igv(self, mock_subprocess):
-        mock_subprocess.return_value.stdout = STREAMING_READS_CONTENT
+    @responses.activate
+    @mock.patch('seqr.views.apis.igv_api.does_file_exist')
+    @mock.patch('seqr.views.apis.igv_api.subprocess.run')
+    def test_proxy_google_to_igv(self, mock_subprocess, mock_file_exist):
+        mock_subprocess.return_value.stdout = b'token\n'
+        mock_subprocess.return_value.returncode = 0
+        mock_file_exist.return_value = False
+
+        responses.add(responses.GET, 'https://project_a.storage.googleapis.com/sample_1.bai',
+                      stream=True,
+                      body=b'\n'.join(STREAMING_READS_CONTENT), status=206)
 
         url = reverse(fetch_igv_track, args=[PROJECT_GUID, 'gs://project_A/sample_1.bam.bai'])
         self.check_collaborator_login(url)
         response = self.client.get(url, HTTP_RANGE='bytes=100-200')
         self.assertEqual(response.status_code, 206)
-        self.assertListEqual([val for val in response.streaming_content], STREAMING_READS_CONTENT)
-        mock_subprocess.assert_called_with(
-            'gsutil cat -r 100-200 gs://project_A/sample_1.bai',
-            stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
-
-        url = reverse(fetch_igv_track, args=[PROJECT_GUID, 'gs://fc-secure-project_A/sample_1.cram.gz'])
-        response = self.client.get(url)
-        self.assertEqual(response.status_code, 200)
-        self.assertListEqual([val for val in response.streaming_content], STREAMING_READS_CONTENT)
-        mock_subprocess.assert_called_with(
-            'gsutil -u anvil-datastorage cat gs://fc-secure-project_A/sample_1.cram.gz',
-            stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True)
+        self.assertEqual(next(response.streaming_content), b'\n'.join(STREAMING_READS_CONTENT))
+        mock_subprocess.assert_called_with(['gcloud', 'auth', 'print-access-token'], capture_output=True)
+        mock_file_exist.assert_called_with('gs://project_A/sample_1.bam.bai', user=self.collaborator_user)
+        self.assertEqual(responses.calls[0].request.headers.get('Range'), 'bytes=100-200')
+        self.assertEqual(responses.calls[0].request.headers.get('Authorization'), 'Bearer token')
+
+        responses.replace(responses.GET, 'https://project_a.storage.googleapis.com/sample_1.bai',
```

Comment:
you deleted us updating the url to `url = reverse(fetch_igv_track, args=[PROJECT_GUID, 'gs://fc-secure-project_A/sample_1.cram.gz'])`

I would still use that different URL here, although its okay to have it return an error. We want to test that its not checking if the `bai` file exists